pkg/controller/atlasfederatedauth/atlasfederated_auth_controller.go

```diff
@@ -114,13 +112,11 @@ func (r *AtlasFederatedAuthReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return result.ReconcileResult(), nil
 	}
 
-	result = r.ensureFederatedAuth(workflowCtx, fedauth)
-	if !result.IsOk() {
-		workflowCtx.SetConditionFromResult(status.FederatedAuthReadyType, result)
-	}
-	workflowCtx.SetConditionTrue(status.ReadyType)
+	result = r.ensureFederatedAuth(workflowCtx, &fedauth)
+	workflowCtx.SetConditionFromResult(status.FederatedAuthReadyType, result)
```

Comment:
note: this changes behavior. Previously we only set `FederatedAuthReady` if the result from `ensure...` is not ok. While this looks like a proper fix, it's something we should be aware about.

Reply:
The conditional was redundant, take a look at serConditionFromResult